### ISSUE TITLE
Randomly skipping and repeating songs

### ISSUE THREAD

**Issue by LiftedStarfish** [aquelemiguel/parrot#241]
### 📝 Description

Sometimes, when switching to the next song in the queue, the app instead decides to replay the previous song. The queue *thinks* its playing the song that's in the queue, but the audio is just a repeat of the previous song. As a result, it ends up skipping the song that's actually in that position in the queue.

### 🪜 Reproduction Steps

I have not been able to find a way to reliably replicate this behavior. What I *can* say is that it's not caused by the queued song being unable to play, because if I queue the song again, it *will* play.

### ℹ Environment / Computer Info

- **Parrot version**: `1.6.0`
- **Operating System**: `Debian GNU/Linux 11 (bullseye) x86_64`
- **Kernel**: `5.10.0-21-amd64`

**Comment by aquelemiguel** [comment]
Hi, @LiftedStarfish!

Good to have you back.

Do you think [this Songbird issue](https://github.com/serenity-rs/songbird/issues/178) could be related?
It does seem awfully similar. They're on Songbird `0.3.2` as well.

**Comment by LiftedStarfish** [comment]
Yes, this sounds like it could be related. Just to be clear, in this case, the current track does not _stop_ playing. It's just that when the next track goes to play, it just replays the previous track, before continuing on with the track after it.

For example, if the queue looks like this:
- Song 1
- Song 2
- Song 3

When the bug occurs, it will play like this:
- Song 1
- Song 1
- Song 3

**Comment by aquelemiguel** [comment]
Understood, I'll try to replicate on my side when I get the chance and keep a close eye on Songbird's issue.
Thank you again for reporting this. ⭐️

**Comment by phoxwupsh** [comment]
@aquelemiguel it seems like this is caused by yt-dlp. I update my yt-dlp version to `2023.04.27.022240`, and the problem is solved, maybe you can give it a try!
